nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/test/java/org/nuxeo/ecm/automation/core/test/CoreOperationsTest.java

```diff
@@ -650,6 +652,31 @@ public void testCreateVersion3() throws Exception {
         }
     }
 
+    @Test
+    public void testCheckinVersion() throws Exception {
+        List<String> expected = List.of("none;0.0", "minor;0.1", "major;1.0");
+        for (String entry : expected) {
+            String[] split = entry.split(";");
+            try (OperationContext ctx = new OperationContext(session)) {
+                ctx.setInput(src);
+
+                Expression expr = Scripting.newExpression("Document.versionLabel");
+                OperationChain chain = new OperationChain("testChain");
+                chain.add(FetchContextDocument.ID);
+                chain.add(CreateDocument.ID)
+                     .set("type", "File")
+                     .set("name", "file")
+                     .set("properties", "dc:title=MyDoc");
+                chain.add(CheckInDocument.ID)
+                     .set("version", split[0])
+                     .set("comment", "yo")
+                     .set("versionVarName", "ver");
+                DocumentModel doc = (DocumentModel) service.run(ctx, chain);
+                assertEquals(split[1], doc.getVersionLabel());
+            }
+        }
+    }
+
```

Comment:
I wonder if could not simply test the single `CheckInDocument.ID` operation, without using a chain (and by the way the scripting expression is not used here).